### PR TITLE
Extraction from main cloud

### PR DESCRIPTION
**Pull request by TheMeinerLP**

### CODE REVIEW

File: chat-plugin/src/main/java/eu/cloudnetservice/chatsystem/ChatPlugin.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    @Override
+    public void onEnable() {
+        getLogger().config("Initiate Bukkit Chat System");
+        getConfig().options().copyDefaults(true);
+        saveConfig();
+        getServer().getPluginManager().registerEvents(new ChatListener(), this);
+    }
+
+    @Override
+    public void onDisable() {
```
Comment (TheMeinerLP) [inline chat-plugin/src/main/java/eu/cloudnetservice/chatsystem/ChatPlugin.java]: Can i remove it or what can do ?

**Comment by TheMeinerLP** [comment]
I think is ready to merge. A have reviewed so code self again. And i found little improvments but not many.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      }
+    }
+    stage('Release ZIP') {
+      steps {
+        sh '''mkdir -p temp;
+        cp chat-plugin/target/*.jar temp/;'''
+        zip archive: true, dir: 'temp', glob: '', zipFile: 'ChatSystem.zip'
+        sh 'rm -r temp/';
+      }
+    }
```
Comment (GiantTreeLP) [inline Jenkinsfile]: Is a ZIP file necessary?

File: chat-plugin/pom.xml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    <artifactId>guava</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>net.md-5</groupId>
+            <artifactId>bungeecord-bootstrap</artifactId>
+            <version>${dependency.bungeecord.version}</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (GiantTreeLP) [inline chat-plugin/pom.xml]: Is a dependency on the full binary of bungeecord necessary?

File: chat-plugin/src/main/java/eu/cloudnetservice/chatsystem/listener/ChatListener.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                                                                                                                e.getMessage()
+                                                                                                                 .replace("%",
+                                                                                                                          "%%")) : ChatColor
+                                                                                  .stripColor(ChatColor.translateAlternateColorCodes('&',
+                                                                                                                                     e.getMessage()
+                                                                                                                                      .replace(
+                                                                                                                                          "%",
+                                                                                                                                          "%%"))))));
+    }
+}
```
Comment (GiantTreeLP) [inline chat-plugin/src/main/java/eu/cloudnetservice/chatsystem/listener/ChatListener.java]: Please reformat this file

File: chat-plugin/src/main/resources/plugin.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+name: ChatPlugin
+main: eu.cloudnetservice.chatsystem.ChatPlugin
+version: ${chat.version}
+
+author: TheMeinerLP
```
Comment (GiantTreeLP) [inline chat-plugin/src/main/resources/plugin.yml]: Please change the author

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+
+        <plugin.shade.version>3.2.1</plugin.shade.version>
+
+        <chat.version>1.0-SNAPSHOT</chat.version>
+
+        <dependency.cloudnet.version>2.1.17</dependency.cloudnet.version>
+        <dependency.spigot.version>1.8.8-R0.1-SNAPSHOT</dependency.spigot.version>
```
Comment (GiantTreeLP) [inline pom.xml:26]: Given the api-version of this plugin, think about the minimum supported spigot version.

File: readme.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+# Chat System | The Cloud Network Environment Technology 2
+![Image of CloudNet](https://cdn.discordapp.com/attachments/325383142464552972/354670548292206594/CloudNet.png)
+
+This is the SimpleNametag System for CloudNet 2.2.0
+ 
+
+ ### Support
+ 
+  * Spigot-Support » 1.8 - 1.15.X
```
Comment (GiantTreeLP) [inline readme.md:21]: Given the api-version in the plugin.yml, this seems incorrect.

**Review by GiantTreeLP — CHANGES_REQUESTED** [review]

File: readme.md (hunk 7 above)
Comment (TheMeinerLP) [reply readme.md:21]: Is correct based on https://www.spigotmc.org/wiki/plugin-yml/ wiki page

File: pom.xml (hunk 6 above)
Comment (TheMeinerLP) [reply pom.xml:26]: Is not necessary, the api-version says only the backward compatibility for spigot 1.13 above

Comment (einrobin) [reply pom.xml:26]: I think it's in the spigot.yml, but you can specify that spigot won't load plugins with an API version lower than X.

File: chat-plugin/src/main/resources/plugin.yml (hunk 5 above)
Comment (TheMeinerLP) [reply chat-plugin/src/main/resources/plugin.yml]: Is changed in the next commits

File: chat-plugin/src/main/java/eu/cloudnetservice/chatsystem/listener/ChatListener.java (hunk 4 above)
Comment (TheMeinerLP) [reply chat-plugin/src/main/java/eu/cloudnetservice/chatsystem/listener/ChatListener.java]: Is formatted in the next commits

File: chat-plugin/pom.xml (hunk 3 above)
Comment (TheMeinerLP) [reply chat-plugin/pom.xml]: Is not, is removed

File: Jenkinsfile (hunk 2 above)
Comment (TheMeinerLP) [reply Jenkinsfile]: Is removed for the final step

**Pull request merged**